mypy/newsemanal/semanal.py

```diff
@@ -867,6 +867,17 @@ def analyze_class(self, defn: ClassDef) -> None:
                 self.prepare_class_def(defn, info)
             return
 
+        is_named_tuple, info = self.named_tuple_analyzer.analyze_namedtuple_classdef(defn)
+        if is_named_tuple:
+            if info is None:
```

Comment:
I'd suggest moving the new code to a separate method, as the method is already quite long.